subprojects/docs-asciidoctor-extensions-base/src/test/groovy/org/gradle/docs/asciidoctor/IdGeneratorTest.groovy

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.docs.asciidoctor
+
+import spock.lang.Specification
+
+class IdGeneratorTest extends Specification {
+
+    def "converts string to identifier"(String source, String id) {
+        expect:
+        IdGenerator.generateId(source) == id
+
+        where:
+        source | id
```

Comment:
🧐 We usually format these data tables, such that column-separator lines are aligned. Reformatting the file in the IDE should do the trick.